jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3VirtualBucket.kt

```diff
@@ -88,7 +94,20 @@ class S3VirtualBucket(private val s3Bucket: Bucket, val client: S3Client) : Ligh
         it.versionId(versionId)
     }
 
-    private companion object {
+    fun deletedBucketErrorHandling() {
+        notifyError(project = project, content = message("s3.open.viewer.bucket_does_not_exist"))
+        val fileEditorManager = FileEditorManager.getInstance(project)
+        fileEditorManager.openFiles.forEach {
+            if (it is S3VirtualBucket && it.name == s3Bucket.name()) {
+                runInEdtAndWait {
```

Comment:
`runInEdtAndWait` is marked `@Testonly` so we have started using coroutines for this